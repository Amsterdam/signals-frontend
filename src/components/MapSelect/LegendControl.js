const LegendControl = L.Control.extend({
  options: {
    elements: []
  },

  initialize(options) {
    L.setOptions(this, options);
  },

  onAdd() {
    const div = L.DomUtil.create('div', 'legend-control');

    const header = L.DomUtil.create('div', 'legend-header', div);
    header.innerText = 'Legenda';

    const content = L.DomUtil.create('div', 'legend-content', div);
    this.options.elements.forEach((element) => {
      const item = L.DomUtil.create('div', 'legend-content-item', content);
      const icon = L.DomUtil.create('img', 'legend-content-icon', item);
      icon.src = element.iconUrl;
      const text = L.DomUtil.create('span', 'legend-content-text', item);
      text.innerText = element.label;
    });

    return div;
  }
});

const constructor = (opts) => new LegendControl(opts);
export default constructor;
