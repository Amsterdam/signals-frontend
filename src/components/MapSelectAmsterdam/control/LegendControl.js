const LegendControl = L.Control.extend({
  options: {
    startClosedWidth: 600,
    elements: [],
  },

  initialize(options) {
    L.setOptions(this, options);
  },

  onAdd(map) {
    this.isClosed = false;

    if (map.getSize().x < this.options.startClosedWidth) {
      this.isClosed = true;
    }

    const div = L.DomUtil.create('div', 'legend-control');

    const header = L.DomUtil.create('div', 'legend-header', div);
    header.innerText = 'Legenda';

    header.addEventListener('click', () => {
      this.isClosed = !this.isClosed;
      this._setClosed(div);
    });
    this._setClosed(div);

    const content = L.DomUtil.create('div', 'legend-content', div);
    this.options.elements.forEach(element => {
      const item = L.DomUtil.create('div', 'legend-content-item', content);
      const icon = L.DomUtil.create('img', 'legend-content-icon', item);
      icon.src = element.iconUrl;
      const text = L.DomUtil.create('span', 'legend-content-text', item);
      text.innerText = element.label;
    });

    return div;
  },

  _setClosed(element) {
    if (this.isClosed) {
      L.DomUtil.addClass(element, 'is-closed');
      L.DomUtil.removeClass(element, 'is-open');
    } else {
      L.DomUtil.removeClass(element, 'is-closed');
      L.DomUtil.addClass(element, 'is-open');
    }
  },
});

export default LegendControl;
