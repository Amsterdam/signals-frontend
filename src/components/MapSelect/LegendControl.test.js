import L from 'leaflet';
import LegendControl from "./LegendControl";

describe('Leaflet legend control', () => {
  let mapDiv;

  const createControl = (options = {}) => {
    mapDiv = document.createElement('div');
    const map = L.map(mapDiv);

    const control = new LegendControl(options);
    control.addTo(map);

    const containerEl = mapDiv.querySelector('.legend-control');
    return [containerEl, control];
  };

  it('is rendered', () => {
    const [containerEl, ] = createControl({
      items: [
        {
          iconUrl: 'foo/bar.svg',
          label: 'bar label',
        }
      ]
    });

    expect(containerEl).toMatchSnapshot();
  });
});
