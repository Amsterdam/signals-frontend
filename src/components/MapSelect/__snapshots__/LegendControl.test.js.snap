// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Leaflet legend control is rendered 1`] = `
<div
  class="legend-control leaflet-control"
>
  <div
    class="legend-header"
  />
  <div
    class="legend-content"
  />
</div>
`;
